ParadiseSS13/Paradise
# Resumeable MODsuit activation

## What Does This PR Do
Adds code to support resuming an interrupted (de)activation of your MODsuit.

Fixes #22834

The `/obj/item/mod/control/proc/toggle_activate` method processes activating your MODsuit. There are 4 different if statements that make changes to your boots, gauntlets, chestplate, and helmet, and then a final if statement that finalizes the activation. They each include a 2 second delay by default by using `/proc/do_after`

Since `do_after` loops and breaks if `user.stat` is truthy, the user being unconscious will make `user.stat` evaluate to 1 and do_after will return false, interrupting the part sealing.

This PR adds a check that the part is not already sealed before calling do_after. If a part is already sealed then no delay occurs, and the next step happens. There is always a final do_after call that will call finish_activation, but you won't have to seal any parts again. (If you are deactivating the suit we check the part is not already deactivated)





## Why It's Good For The Game

I think it doesn't make sense that a part would have to be sealed again, after it already displays the text "X wraps around your skin", and sounds annoying

## Testing


1. Equip captain's MODsuit
2. Deploy
3. Start activation
4. Wait until 1 or 2 parts are sealed
5. Use right click variable editor to give myself 500 damage (Dies)
6. Use right click rejuvenate 
7. Start activation
8. See that the 1 or 2 parts that were already sealed were skipped
9. See that the activation process was completed normally

The deactivation process was tested similarly.

Note: I think if the skin was changed this might break, since it reads the skin name, instead of a is_sealed variable, but I don't think any code is changing the skin at the moment, or if you did change the skin I think you'd have to remove the suit first?



## Changelog
:cl:
fix: Fix activating MODsuits while falling unconscious a bunch to save activation progress
/:cl:
